gammapy/estimators/energydependence.py

```diff
@@ -0,0 +1,250 @@
+        self.model = model
+        self.num_energy_bands = len(self.energy_edges) - 1
+
+        if fit is None:
+            fit = Fit(optimize_opts={"print_level": 1})
+
+        self.fit = fit
+        self.selection_optional = selection_optional
+
+    def estimate_source_significance(self, dataset):
+        """Estimate the significance of the source above the background.
+
+        Parameters
+        ----------
+        dataset : `~gammapy.datasets.MapDataset`
+            Input dataset to use.
+
+        Returns
+        -------
+        table_bkg_src : `~astropy.table.Table`
+            Table with the results of the null hypothesis with no source, and alternative
+            hypothesis with the source added in. Entries are:
+            * "Emin" : the minimum energy of the energy band
+            * "Emax" : the maximum energy of the energy band
+            * "delta ts" : difference in ts
+            * "df" : the degrees of freedom between null and alternative hypothesis
+            * "significance" : significance of the result
+        """
+
+        dataset.mask_fit = dataset.counts.geom.energy_mask(
+            energy_min=self.energy_edges[0], energy_max=None
+        )
+
+        # Calculate the dataset for each energy slice
+        slices_src = Datasets()
+        for emin, emax in zip(self.energy_edges[:-1], self.energy_edges[1:]):
+            sliced_src = dataset.slice_by_energy(emin, emax)
+            bkg_sliced_model = FoVBackgroundModel(dataset_name=sliced_src.name)
+            sliced_src.models = [self.model.copy(), bkg_sliced_model]
+            slices_src.append(sliced_src)
+
+        # Norm is free and fit
+        test_results = []
+        for sliced in slices_src:
+            parameters = [param for param in sliced.models.parameters.free_parameters]
+            null_values = [0] + [
+                param.value
+                for param in sliced.models[0].spatial_model.parameters.free_parameters
+            ]
+
+            test = TestStatisticNested(
+                parameters=parameters,
+                null_values=null_values,
+                n_sigma=-np.inf,
+                fit=self.fit,
+            )
+            test_results.append(test.run(sliced))
+
+        delta_ts_bkg_src = [_["ts"] for _ in test_results]
+        df_src = np.sum(
+            [
+                len(_["fit_results"].parameters.free_parameters.names)
+                for _ in test_results
+            ]
+        )
+        df_bkg = 1 * self.num_energy_bands
+        df_bkg_src = df_src - df_bkg
+        sigma_ts_bkg_src = ts_to_sigma(delta_ts_bkg_src, df=df_bkg_src)
+
+        # Prepare results dictionary for signal above background
+        result_bkg_src = {}
+
+        result_bkg_src["Emin"] = self.energy_edges[:-1]
+        result_bkg_src["Emax"] = self.energy_edges[1:]
+        result_bkg_src["delta ts"] = [val for val in delta_ts_bkg_src]
+        result_bkg_src["df"] = [df_bkg_src] * self.num_energy_bands
```

Comment:
this I don't understand. Running your notebook, I get `df = [12, 12, 12]`.
It should be (3+2 = 5) in each bin, no?